### PR TITLE
Add Input Consistency Check

### LINKED ISSUE #3306: Input consistency check.
We need some way to check if input scenario (config + plans + other input files) are consistent, because sometimes they are not.
We have src/main/scala/beam/utils/ConfigConsistencyComparator.scala but it checks only config file, not plans and other files.

Right now we need to check if the values from 
```
beam.agentsim.agents.rideHail.initialization.procedural.vehicleTypeId
beam.agentsim.agents.vehicles.dummySharedCar.vehicleTypeId
```
are present in vehicle types from file 
```
beam.agentsim.agents.vehicles.vehicleTypesFilePath
```

### PR DESCRIPTION
**Pull request by alex-vv**
Adds basic input consistency check, currently only checking vehicle types as described in the ticket.

Fixes #3306

---
This change is [Reviewable](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3318)

### CODE REVIEW

**Comment by alex-vv** [comment]
test!

**Comment by alex-vv** [comment]
test!

File: src/main/scala/beam/sim/BeamHelper.scala [hunk 1]
```diff
@@ -893,6 +893,11 @@ trait BeamHelper extends LazyLogging {
     logger.debug(s"Beam output directory is: $outputDirectory")
     logger.info(ConfigConsistencyComparator.getMessage.getOrElse(""))
 
+    val errors = InputConsistencyCheck.checkConsistency(beamConfig)
+    if (errors.nonEmpty) {
+      logger.error("Input consistency check failed:\n" + errors.mkString("\n"))
```
Comment (JustinPihony) [inline src/main/scala/beam/sim/BeamHelper.scala:898]: Should this fail?

File: src/main/scala/beam/utils/InputConsistencyCheck.scala [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    */
+  def checkVehicleTypes(
+    vehicleTypes: Set[Id[BeamVehicleType]],
+    rideHailTypeId: String,
+    dummySharedCarTypeId: String
+  ): List[String] = {
+    val errors = ListBuffer[String]()
+    val stringTypes = vehicleTypes.map(_.toString)
+    if (!stringTypes.contains(rideHailTypeId)) {
+      errors.append("beam.agentsim.agents.rideHail.initialization.procedural.vehicleTypeId is not in vehicleTypes")
```
Comment (JustinPihony) [inline src/main/scala/beam/utils/InputConsistencyCheck.scala]: To better help debug the error this should list the actual id found. And also should probably list the types tried maybe?

File: src/main/scala/beam/sim/BeamHelper.scala (hunk 1 above)
Comment (alex-vv) [reply src/main/scala/beam/sim/BeamHelper.scala:898]: probably, @nikolayilyin should it fail in that case?

File: src/main/scala/beam/utils/InputConsistencyCheck.scala (hunk 2 above)
Comment (alex-vv) [reply src/main/scala/beam/utils/InputConsistencyCheck.scala]: done

File: src/main/scala/beam/sim/BeamHelper.scala (hunk 1 above)
Comment (nikolayilyin) [reply src/main/scala/beam/sim/BeamHelper.scala:898]: yes, this should fail, otherwise it might fail somewhere during simulation

**Comment by alex-vv** [comment]
test!

**Comment by alex-vv** [comment]
test!

**Comment by alex-vv** [comment]
test!

**Review by nikolayilyin — APPROVED** [review]
LGTM

**Comment by alex-vv** [comment]
test!

**Comment by alex-vv** [comment]
test!

**Comment by alex-vv** [comment]
test!

**Comment by alex-vv** [comment]
test!

**Comment by alex-vv** [comment]
test!

**Pull request merged**